{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Preparation of CCLE data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "from matplotlib import pyplot as plt\n",
    "from pathlib import Path\n",
    "import janitor\n",
    "\n",
    "plt.style.use('seaborn-whitegrid')\n",
    "plt.rcParams['figure.figsize'] = (8.0, 5.0)\n",
    "plt.rcParams['axes.titlesize'] = 18\n",
    "plt.rcParams['axes.labelsize'] = 15\n",
    "\n",
    "data_dir = Path('../data')\n",
    "modeling_data_dir = Path('../modeling_data')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/jc604/.conda/envs/demeter2-stan/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3146: DtypeWarning: Columns (25,26,27,31) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>ncbi_build</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>strand</th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>reference_allele</th>\n",
       "      <th>...</th>\n",
       "      <th>iscosmichotspot</th>\n",
       "      <th>cosmichscnt</th>\n",
       "      <th>exac_af</th>\n",
       "      <th>wes_ac</th>\n",
       "      <th>sangerwes_ac</th>\n",
       "      <th>sangerrecalibwes_ac</th>\n",
       "      <th>rnaseq_ac</th>\n",
       "      <th>hc_ac</th>\n",
       "      <th>rd_ac</th>\n",
       "      <th>wgs_ac</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>375790</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>979072</td>\n",
       "      <td>979072</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27:24</td>\n",
       "      <td>9:10</td>\n",
       "      <td>9:12</td>\n",
       "      <td>104:20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15:13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATAD3A</td>\n",
       "      <td>55210</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>1459233</td>\n",
       "      <td>1459233</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>29:49</td>\n",
       "      <td>33:40</td>\n",
       "      <td>30:38</td>\n",
       "      <td>315:308</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17:31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NADK</td>\n",
       "      <td>65220</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>1685635</td>\n",
       "      <td>1685635</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25:39</td>\n",
       "      <td>16:19</td>\n",
       "      <td>17:20</td>\n",
       "      <td>176:266</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14:23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PLCH2</td>\n",
       "      <td>9651</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>2436128</td>\n",
       "      <td>2436128</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9:20</td>\n",
       "      <td>19:22</td>\n",
       "      <td>20:20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23:15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LRRC47</td>\n",
       "      <td>57470</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>3703695</td>\n",
       "      <td>3703695</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>19:21</td>\n",
       "      <td>7:19</td>\n",
       "      <td>8:17</td>\n",
       "      <td>87:104</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11:16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  hugo_symbol  entrez_gene_id  ncbi_build chromosome  start_position  \\\n",
       "0        AGRN          375790          37          1          979072   \n",
       "1      ATAD3A           55210          37          1         1459233   \n",
       "2        NADK           65220          37          1         1685635   \n",
       "3       PLCH2            9651          37          1         2436128   \n",
       "4      LRRC47           57470          37          1         3703695   \n",
       "\n",
       "   end_position strand variant_classification variant_type reference_allele  \\\n",
       "0        979072      +                 Silent          SNP                A   \n",
       "1       1459233      +                 Silent          SNP                A   \n",
       "2       1685635      +      Missense_Mutation          SNP                G   \n",
       "3       2436128      +      Missense_Mutation          SNP                G   \n",
       "4       3703695      +                 Silent          SNP                G   \n",
       "\n",
       "   ... iscosmichotspot  cosmichscnt   exac_af wes_ac sangerwes_ac  \\\n",
       "0  ...           False            0       NaN  27:24         9:10   \n",
       "1  ...           False            0  0.000008  29:49        33:40   \n",
       "2  ...           False            0       NaN  25:39        16:19   \n",
       "3  ...           False            0       NaN   9:20        19:22   \n",
       "4  ...           False            0  0.000033  19:21         7:19   \n",
       "\n",
       "  sangerrecalibwes_ac rnaseq_ac hc_ac rd_ac  wgs_ac  \n",
       "0                9:12    104:20   NaN   NaN   15:13  \n",
       "1               30:38   315:308   NaN   NaN   17:31  \n",
       "2               17:20   176:266   NaN   NaN   14:23  \n",
       "3               20:20       NaN   NaN   NaN   23:15  \n",
       "4                8:17    87:104   NaN   NaN   11:16  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutation_df = pd.read_csv(data_dir / 'CCLE_mutation_data.csv').clean_names()\n",
    "mutation_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "mutation_columns = [\n",
    "    'tumor_sample_barcode', 'hugo_symbol', 'chromosome', 'start_position', \n",
    "    'end_position', 'variant_classification', 'variant_type', 'protein_change'\n",
    "]\n",
    "mutation_df = mutation_df[mutation_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_change</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell_line</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A427_LUNG</th>\n",
       "      <td>p.G12D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A549_LUNG</th>\n",
       "      <td>p.G12S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AGS_STOMACH</th>\n",
       "      <td>p.G12D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMO1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <td>p.A146T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASPC1_PANCREAS</th>\n",
       "      <td>p.G12D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SW948_LARGE_INTESTINE</th>\n",
       "      <td>p.Q61L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCCPAN2_PANCREAS</th>\n",
       "      <td>p.G12R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TOV21G_OVARY</th>\n",
       "      <td>p.G13C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UMUC3_URINARY_TRACT</th>\n",
       "      <td>p.G12C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YD8_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <td>p.G138V</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>114 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        protein_change\n",
       "cell_line                                             \n",
       "A427_LUNG                                       p.G12D\n",
       "A549_LUNG                                       p.G12S\n",
       "AGS_STOMACH                                     p.G12D\n",
       "AMO1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE        p.A146T\n",
       "ASPC1_PANCREAS                                  p.G12D\n",
       "...                                                ...\n",
       "SW948_LARGE_INTESTINE                           p.Q61L\n",
       "TCCPAN2_PANCREAS                                p.G12R\n",
       "TOV21G_OVARY                                    p.G13C\n",
       "UMUC3_URINARY_TRACT                             p.G12C\n",
       "YD8_UPPER_AERODIGESTIVE_TRACT                  p.G138V\n",
       "\n",
       "[114 rows x 1 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hotspot_codons = [12, 13, 59, 61, 146]\n",
    "hotspot_codons_re = '12|13|59|61|146'\n",
    "\n",
    "kras_mutations = mutation_df \\\n",
    "    .pipe(lambda x: x[x.hugo_symbol == 'KRAS']) \\\n",
    "    .pipe(lambda x: x[x.variant_classification == 'Missense_Mutation']) \\\n",
    "    .pipe(lambda x: x[x.protein_change.str.contains(hotspot_codons_re)]) \\\n",
    "    [['tumor_sample_barcode', 'protein_change']] \\\n",
    "    .drop_duplicates() \\\n",
    "    .rename({'tumor_sample_barcode': 'cell_line'}, axis=1) \\\n",
    "    .groupby('cell_line') \\\n",
    "    .aggregate(lambda x: ';'.join(x.protein_change))\n",
    "kras_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_change</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell_line</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NCIH2291_LUNG</th>\n",
       "      <td>p.G12V;p.G12C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              protein_change\n",
       "cell_line                   \n",
       "NCIH2291_LUNG  p.G12V;p.G12C"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kras_mutations[kras_mutations.protein_change.str.contains(';')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "mutation_df.to_csv(modeling_data_dir / 'ccle_mutations.csv', index=False)\n",
    "kras_mutations.to_csv(modeling_data_dir / 'kras_mutants.csv', index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
